src/modules/mavlink/mavlink_params.c

```diff
@@ -155,3 +164,66 @@ PARAM_DEFINE_INT32(MAV_HB_FORW_EN, 1);
  * @max 250
  */
 PARAM_DEFINE_INT32(MAV_RADIO_TOUT, 5);
+
+/**
+ * First 4 bytes of secret key for mavlink
+ *
+ * @group MAVLink
+ */
+PARAM_DEFINE_INT32(MAV_SIGN_KEY1, -835092985);
```

Comment:
So this key gets written in from the signing message right, and that can only be sent over the UART? 

Don't all params get sent when you connect to a GCS? I mean doesn't this mean the key in the params will be sent in plaintext when you first connect your GCS?

Reply:
Ie. is there a secure part this can be stored in?

Reply:
@hamishwillee 
1) SIGNING_MESSAGE can be sent over UART only , correct
2) Yes all params are sent to GCS. However you have a MAV_SIGN_MODE param which defines when the messages should be hashed and when its optional

Therefore the recommended flow is 
1) Connect PX4 to GCS ( or any other control ) 
2) Configure the keys and set SIGN_MODE to something more secure ( either allow unsigned messages over UART only or block them at all ) 
3) after this stage only hashed messages are allowed to flow either for all channels or for non UART channels

Reply:
@hamishwillee if you plan to work with GCS over the air then you are correct it will expose all the key pieces and timestamp. 
In our scenario we dont do param_list over the air , and therefore they are not exposed. 
Is there a way to exclude specific parameters to be available over mavlink on px4 ( e.g hide parameters from GCS )?

Reply:
@hamishwillee do you have an example how to write to place other then parameters, including the sdcard ? from documentation it looks like the params are the only option

Reply:
There's no way to "hide" parameters but there should be. This has been asked about for years -- Remote ID parameters need this as well to meet the "tamper proof" requirement.

Reply:
So if someone turns the device on and off again it will accept any packet? Or will it ignore the packet because it doesn't have the matching key?

The key should be stored in persistent storage as per  https://mavlink.io/en/guide/message_signing.html#secret_key

My problem here though is that there doesn't seem to be any way for the flight stack to indicate that setting the key failed. So I want to say "make it fail if there is no SD card" and emit a critical event". Doable?